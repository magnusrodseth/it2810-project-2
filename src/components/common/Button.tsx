import React, { ReactNode } from 'react'
import { Button as DaisyButton } from 'react-daisyui'
import { ButtonProps as DaisyButtonProps } from 'react-daisyui/dist/Button'
import classNames from '../../utils/common/classNames'

interface ButtonProps extends DaisyButtonProps {
  children: ReactNode
  onClick?: () => void
  dataTestId?: string
}

const Button: React.FC<ButtonProps> = ({ children, onClick, color, className, dataTestId }) => {
  const style = className || ''

  return (
    <DaisyButton
      onClick={onClick}
      className={classNames('py-2 px-4', style)}
      color={color}
      data-testid={dataTestId}
    >
      {children}
    </DaisyButton>
  )
}

export default Button
