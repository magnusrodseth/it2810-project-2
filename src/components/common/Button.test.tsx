import renderer from 'react-test-renderer'
import Button from './Button'

describe('Button', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Button onClick={() => {}}>Hello</Button>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
