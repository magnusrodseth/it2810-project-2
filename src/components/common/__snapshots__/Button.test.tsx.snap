// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="btn py-2 px-4"
  onClick={[Function]}
>
  Hello
</button>
`;
